### DISCUSSION TITLE
20k Local DOFs - does this include Auxiliary system?

### DISCUSSION THREAD

**Discussion by ln53** [idaholab/moose#18879] [Q&A General] [answered]
Hi,
In the documentation [here](https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step07_parallel.html), the recommended target for DOFs-per-process is 20,000. Is this for just the Nonlinear System or should this include the Auxiliary System DOFs as well?
Thanks

**Comment by roystgnr** [comment, marked as answer]
Just for the nonlinear system.

It's also a "loose enough" number that I wouldn't worry about missing it by a factor of 2 (as you would if you included the auxiliary system in an average problem).  It's just a rule of thumb, and one that varies from system to system (relatively faster interconnects mean you can scale better to higher processor counts with fewer DoFs-per-process; relatively faster CPU cores mean you want to give each more work rather than having it wait on network latency).  Nothing's likely to break if you use too many or too few DoFs per core, you just might be missing out on or being inefficient with parallel speedup, respectively.

**Reply by ln53** [reply]
Thanks for the answer and additional insight!
